torax/_src/sources/impurity_radiation_heat_sink/impurity_radiation_constant_fraction.py

```diff
@@ -57,7 +58,7 @@ def radially_constant_fraction_of_Pin(
 
   # TODO(b/383061556) Move away from using brittle source names to identify
   # sinks/sources.
-  source_profiles = jnp.zeros_like(geo.rho)
+  source_profiles = jnp.zeros_like(geo.rho, dtype=jax_utils.get_dtype())
```

Comment:
are these changes needed? Granted they are positive changes, was just curious!

Reply:
They are not!